src/core/providers/ogr/qgsogrprovidermetadata.cpp

```diff
@@ -1071,6 +1075,118 @@ bool QgsOgrProviderMetadata::saveLayerMetadata( const QString &uri, const QgsLay
   throw QgsNotSupportedException( QObject::tr( "Storing metadata for the specified uri is not supported" ) );
 }
 
+QList<QgsLayerMetadataProviderResult> QgsOgrProviderMetadata::searchLayerMetadata( const QString &uri, const QString &searchString, const QgsRectangle &geographicExtent, QgsFeedback *feedback )
+{
+
+  QList<QgsLayerMetadataProviderResult> results;
+  QgsProviderMetadata *md { QgsProviderRegistry::instance()->providerMetadata( QStringLiteral( "ogr" ) ) };
+  Q_ASSERT( md );
+  std::unique_ptr<QgsAbstractDatabaseProviderConnection> conn;
+  conn.reset( static_cast<QgsAbstractDatabaseProviderConnection *>( md->createConnection( uri, {} ) ) );
+  if ( conn && ( ! feedback || ! feedback->isCanceled() ) )
+  {
+    try
+    {
+      const QString searchQuery { QStringLiteral( R"SQL(
+      SELECT
+        ref.table_name, md.metadata, gc.geometry_type_name
+      FROM
+        gpkg_metadata_reference AS ref
+      JOIN
+        gpkg_metadata AS md ON md.id = ref.md_file_id
+      JOIN
+        gpkg_geometry_columns AS gc ON gc.table_name = ref.table_name
+      WHERE
+        md.md_standard_uri = 'http://mrcc.com/qgis.dtd'
+        AND ref.reference_scope = 'table'
+        AND md.md_scope = 'dataset'
+      )SQL" ) };
+
+      const QList<QVariantList> cMetadataResults { conn->executeSql( searchQuery, feedback ) };
+      for ( const QVariantList &mdRow : std::as_const( cMetadataResults ) )
+      {
+
+        if ( feedback && feedback->isCanceled() )
+        {
+          break;
+        }
+
+        // Read MD from the XML
+        QDomDocument doc;
+        doc.setContent( mdRow[1].toString() );
+        QgsLayerMetadata layerMetadata;
+        if ( layerMetadata.readMetadataXml( doc.documentElement() ) )
+        {
+          QgsLayerMetadataProviderResult result;
+          result.metadata = layerMetadata;
+
+          QgsRectangle extents;
+
+          const auto cExtents { layerMetadata.extent().spatialExtents() };
+          for ( const auto &ext : std::as_const( cExtents ) )
+          {
+            QgsRectangle bbox {  ext.bounds.toRectangle()  };
+            QgsCoordinateTransform ct { ext.extentCrs, QgsCoordinateReferenceSystem::fromEpsgId( 4326 ), QgsProject::instance()->transformContext() };
```

Comment:
No QgsProject::instance() please! You may need to create a "metadata search context" to pass with the transform context instead (nicer), or add a compulsory QgsTransformContext argument to searchLayerMetadata (not as nice)